Makefile.package

```diff
@@ -151,9 +160,28 @@ info-channel:
 info-fqn:
 	echo ${PACKAGE_FQN}
 
+info: parse-info
+	echo ${PACKAGE_INFO}
+
 # CONFIGURATION TARGETS -------------------------------------------------------
 .PHONY: editable uneditable
 
+ifneq "${LOCKFILE_SOURCE}" ""
+LOCKFILE_OPTION := --lockfile="${BUILD_LOCKFILE}"
+${BUILD_LOCKFILE}: ${LOCKFILE_SOURCE}
+	# Create lockfile from LOCKFILE_SOURCE.
+	#
+	mkdir -p "${BUILD_DIR}"
+	conan lock create \
+		--lockfile-out "${BUILD_LOCKFILE}" \
+		--build=${BUILD_POLICY} \
```

Comment:
What is missing here is 
`    --build ${PACKAGE_NAME} \`
Otherwise, the package build requirements will not be part of the lockfile, which leads to errors during package build.

 - [ ] Fix build requirements not being added to lockfile

Reply:
That's odd. I didn't notice a difference whether there or not, but I guess I didn't test widely enough. Thanks for the heads up!

Reply:
I also don't get why we need to specify --build at all in lock file creation. We're not building anything during creation of the lock and it also shouldn't affect dependency resolution, should it?

Reply:
I did some comparison. Adding `--build` seems to only add some *build dependencies*.

Do you have an example where errors occur due to lacking `--build` arguments to lockfile creation?

Reply:
Yes, the missing build dependencies cause an error e.g. if you do 
```
make -C models clean all LOCKFILE_SOURCE=../conanfile.py
ERROR: Build-require 'gtest' cannot be found in lockfile
```
But with your change [here](https://github.com/eclipse/cloe/blob/ben/add-lockfile-option/Makefile.package#L78) that is fixed.